### PR TITLE
Increase the client-go rest client rate limit inside the csi sidecar components

### PR DESCRIPTION
**Pull request by PhanLe1010**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#8699

#### What this PR does / why we need it:

Unblock the Longhorn control plane when there is a surge of volume creation/attach/detach/snapshotting/resizer. The CSI component will process the events faster.

#### Additional documentation or context
See more details about reducing steps in the issue description longhorn/longhorn#8699

### CODE REVIEW

File: csi/deployment.go [hunk 1]
```diff
@@ -67,6 +67,8 @@ func NewAttacherDeployment(namespace, serviceAccount, attacherImage, rootDir str
 			"--timeout=1m50s",
 			"--leader-election",
 			"--leader-election-namespace=$(POD_NAMESPACE)",
+			"--kube-api-qps=50",
+			"--kube-api-burst=100",
```
Comment (shuo-wu) [inline csi/deployment.go:71]: NIT: Use constants rather than hard-coded values

**Review by shuo-wu — DISMISSED** [review]
LGTM

File: csi/deployment.go (hunk 1 above)
Comment (PhanLe1010) [reply csi/deployment.go]: Thank you. Fixed

**Review by shuo-wu — APPROVED** [review]

File: types/types.go [hunk 2]
```diff
@@ -204,6 +204,9 @@ const (
 
 	CniNetworkNone          = ""
 	StorageNetworkInterface = "lhnet1"
+
+	KubeApiQps   = 50
+	KubeApiBurst = 100
```
Comment (derekbit) [inline types/types.go:209]: Just curious what will happen if using bigger numbers.

**Review by ejweber — APPROVED** [review]
LGTM.

TBH, I'm not sure how we can decide what the "right" values are here. Presumably, the default values set in the sidecar are intended to prevent the volume provisioning/attaching infrastructure from unintentionally denying service to other Kubernetes API clients. It seems fine to increase them to the values we are already using for other Longhorn components (as you have done here), but there is probably some limit to how high they should go. If we decide we want to increase them further in the future, we will need a plan to figure that limit out.

File: types/types.go (hunk 2 above)
Comment (PhanLe1010) [reply types/types.go:209]: I don't see benefit from increasing it to bigger values when testing scaling yet. These values seems big enough to eliminate the throttling

**Comment by PhanLe1010** [comment]
> TBH, I'm not sure how we can decide what the "right" values are here. Presumably, the default values set in the sidecar are intended to prevent the volume provisioning/attaching infrastructure from unintentionally denying service to other Kubernetes API clients. It seems fine to increase them to the values we are already using for other Longhorn components (as you have done here), but there is probably some limit to how high they should go. If we decide we want to increase them further in the future, we will need a plan to figure that limit out.

As we discussed in the US sync. There isn't a concert rule for determine how big is acceptable. I just pick the value base on:
1. Scale test: these values seems big enough to avoid throttling 
2. Other projects are using similar value: [kubectl](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/#:~:text=%2D%2Devent%2Dqps%20int32%C2%A0%C2%A0%C2%A0%C2%A0%C2%A0Default%3A%2050)

**Comment by PhanLe1010** [comment]
@mergify backport v1.6.x v1.5.x

**Pull request merged**